src/components/listing-components/login-dialog.tsx

```diff
@@ -80,86 +92,202 @@ const loginDialog: React.FC<LoginDialogProps> = ({ open, onClose, onLogin }) =>
   const validateLogin = (usernameS: string, passwordS: string) => {
     return usernameS.trim() !== "" && passwordS.trim() !== "";
   };
-  
+
   /**
    * Handle login
+   * @param event 
    */
-  const handleLogin = () => {
+  const handleLogin = async (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
     const isValidLogin = validateLogin(username, password);
-  
     if (isValidLogin) {
-      onLogin();
-      onClose();
+      const { accessToken: newAccessToken, refreshToken: newRefreshToken, error } = await authenticate({
+        selectedSite: siteList.find(siteItem => siteItem.id === site),
+        username: username,
+        password: password
+      });
+
+      if (error) {
+        setLoginError(error);
+      } else {
+        setAccessToken(newAccessToken);
+        onAccessTokenUpdate(newAccessToken);
+        setRefreshToken(newRefreshToken);
+        setTokenTimestamp(Math.floor(Date.now() / 1000));
+        loggedSite(siteList.find(siteItem => siteItem.id === site));
+        onLogin();
+        onClose(loginError);
+      }
     } else {
       setLoginError(strings.errorHandling.listingScreenLogin.loginFailed);
     }
   };
 
+  /**
+   * Token refresh logic
+   */
+  const tokenRefresh = async () => {
+    const { accessToken: newAccessToken, refreshToken: newRefreshToken, error } = await refreshAuthentication({
+      selectedSite: siteList.find(siteItem => siteItem.id === site),
+      refreshToken: refreshToken
+    });
+    if (error) {
+      errorContext.setError(error);
+    } else {
+      setRefreshToken(newRefreshToken);
+      setTokenTimestamp(Math.floor(Date.now() / 1000));
+      setAccessToken(newAccessToken);
+    }
+  };
+
+  /**
+   * Check if the access token is expiring
+   * @param token The access token object
+   * @returns True if the access token is expiring, false otherwise
+   */
```

Comment:
```suggestion
  /**
   * Check if the access token is expiring
   *
   * @param token The access token object
   * @returns True if the access token is expiring, false otherwise
   */
```